src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
-                                        "When " + CurrentModuleObject + " = " +
-                                            state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum) +
-                                            " is used in AirLoopHVAC:DedicatedOutdoorAirSystem,");
+                                        format("When {} = {} is used in AirLoopHVAC:DedicatedOutdoorAirSystem,", CurrentModuleObject, CompName));
                         ShowContinueError(state,
-                                          " the FAN:CONSTANTVOLUME can not be used as a component. The alllowed fan types are FAN:SYSTEMMODEL and "
+                                          " the FAN:CONSTANTVOLUME can not be used as a component. The allowed fan types are FAN:SYSTEMMODEL and "
                                           "FAN:COMPONENTMODEL. Please change it");
                         errorsFound = true;
-                    } else if (SELECT_CASE_var == "FAN:VARIABLEVOLUME") {
+                        break;
+
+                    case ValidEquipListType::FanVariableVolume:
                         ShowSevereError(state,
-                                        "When " + CurrentModuleObject + " = " +
-                                            state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum) +
-                                            " is used in AirLoopHVAC:DedicatedOutdoorAirSystem,");
+                                        format("When {} = {} is used in AirLoopHVAC:DedicatedOutdoorAirSystem,", CurrentModuleObject, CompName));
                         ShowContinueError(state,
-                                          " the FAN:VARIABLEVOLUME can not be used as a component. The alllowed fan types are FAN:SYSTEMMODEL and "
+                                          " the FAN:VARIABLEVOLUME can not be used as a component. The allowed fan types are FAN:SYSTEMMODEL and "
                                           "FAN:COMPONENTMODEL. Please change it");
                         errorsFound = true;
-                    } else if (SELECT_CASE_var == "FAN:SYSTEMMODEL") {
+                        break;
+
+                    case ValidEquipListType::FanSystemModel:
                         thisDOAS.FanName = CompName;
                         thisDOAS.m_FanTypeNum = SimAirServingZones::CompType::Fan_System_Object;
                         thisDOAS.m_FanIndex = HVACFan::getFanObjectVectorIndex(state, CompName);
-                        state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).InletNodeNum(CompNum) =
-                            state.dataHVACFan->fanObjs[thisDOAS.m_FanIndex]->inletNodeNum;
-                        if (state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).InletNodeNum(CompNum) == 0) {
+                        thisInletNodeNum = state.dataHVACFan->fanObjs[thisDOAS.m_FanIndex]->inletNodeNum;
+                        if (thisInletNodeNum == 0) {
```

Comment:
Can use `InletNodeErrFlag |= (thisInletNodeNum == 0);`

Reply:
So ... 

- `Y = X` is the same as `if(X) Y = true; else Y = false;`
- `Y |= X` is the same as `if(X) Y = true;`
- `Y &= X` is the same as `else Y = false;`